RockPaperScissors/RockPaperScissors/RPSGame.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  RPSGame.swift
+//  RockPaperScissors
+//
+//  Created by 전성수 on 2023/09/06.
+//
+
+import Foundation
+
+class RPSGame {
+    
+    func run() {
+        while true {
+            RPSPrinter.menu.printMessage()
+            
+            guard let userChoice = getUserChoice() else {
+                RPSPrinter.invalid.printMessage()
+                continue
+            }
+            
+            if userChoice == .stop {
+                RPSPrinter.gameFinish.printMessage()
+                break
+            }
+            
+            guard let botChoice = generateRandomChoice() else {
+                RPSPrinter.invalid.printMessage()
+                continue
+            }
+            
+            let result = getResult(userChoice, botChoice)
+            result.printMessage()
+            
+            if getGameResult(result) { break }
+        }
+    }
+    
+    
+    private func getUserChoice() -> Input? {
+        guard let input = readLine(), let number = Int(input), let choice = Input(number), choice != Input.error("Invalid Error: \(number)") else {
```

Comment:
너무길면서 조건문 한줄씩 개행하는게 더 좋습니다 :)

Reply:
```suggestion
        guard let input = readLine(), let number = Int(input), let input = Input(number), input != Input.error("Invalid Error: \(number)") else {
```
=> choice 라는 네이밍은 이제 없어도 될 것 같아요 ㅎㅎ

Reply:
error 인 경우에 다시 재시작하는 로직이라면 guard let 말고 input 이 error 인 경우에 continue 해도 되지 않을까요? 
아니면 
1) input 을 Int 로 치환했을때 nil 이거나 input 이 error 인 경우에는 에러를 던져주고 (throw error),
2) input 이 RPS이거나 stop 인 경우에는 정상적으로 Input을 리턴해줘도 되고요.